numpydoc/tests/test_full.py

```diff
@@ -89,7 +91,14 @@ def test_reference(sphinx_app, html_file, expected_length):
     with open(op.join(out_dir, *html_file)) as fid:
         html = fid.read()
 
-    reference_list = re.findall(r'<a class="fn-backref" href="\#id\d+">(.*)<\/a>', html)
+    # TODO: This check can be removed when the minimum supported docutils version
+    # for numpydoc is docutils>=0.18
+    pattern = (
+        'role="doc-backlink"'
+        if version.parse(docutils_version) >= version.parse("0.18")
```

Comment:
This version check should stay as `0.18` though; the following change came in 0.18b1

```
  - Use ARIA roles to enable accessible HTML for abstract, dedication,
    the table of contents, footnote, references, footnotes, citations,
    and backlinks.
```